### DISCUSSION TITLE
Incorporating individual measurement uncertainties into Gaussian Process

### DISCUSSION THREAD

**Discussion by Paeyus** [cornellius-gp/gpytorch#2501] [General]
I have a set of observations, f_i=f(x_i), and I want to construct a probabilistic surrogate, f(x) ~ N[mu(x), sigma(x)], where N is a normal distribution. Each observed output, f_i, is associated with a measurement uncertainty, sigma_i. I would like to incorporate these measurement uncertainties into my surrogate, f_i, so that mu(x) predicts the observations, f_i(x_i), and that the predicted standard deviation, sigma(x_i), envelops the uncertainty in the observed output, epsilon_i.

**Comment by Balandat** [comment]
If you just want to predict the posterior over the latent `f` (rather than the observations `y`), this is straightforward to do by using a `FixedNoiseGaussianLikelihood` (which takes the tensor of noise levels) instead of a `GaussianLikelihood` (which infers the noise level). If you want to be able to compute the posterior of `y` at arbitrary (unobserved) points you need to do a bit more work and also model the noise function across the domain. You can take a look at BoTorch's [`HeteroskedasticSingleTaskGP`](https://github.com/pytorch/botorch/blob/main/botorch/models/gp_regression.py#L253) model to see how this could work.
